public/app/plugins/datasource/prometheus/querybuilder/QueryPattern.tsx

```diff
@@ -0,0 +1,118 @@
+import { css } from '@emotion/css';
+import React from 'react';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { Button, Card, useStyles2 } from '@grafana/ui';
+import { RawQuery } from 'app/plugins/datasource/prometheus/querybuilder/shared/RawQuery';
+
+import promqlGrammar from '../promql';
+
+import { promQueryModeller } from './PromQueryModeller';
+import { PromQueryPattern } from './types';
+
+type Props = {
+  pattern: PromQueryPattern;
+  hasNewQueryOption: boolean;
+  hasPreviousQuery: boolean;
+  selectedPatternName: string | null;
+  setSelectedPatternName: (name: string | null) => void;
+  onPatternSelect: (pattern: PromQueryPattern, selectAsNewQuery?: boolean) => void;
+};
+
+export const QueryPattern = (props: Props) => {
+  const { pattern, onPatternSelect, hasNewQueryOption, hasPreviousQuery, selectedPatternName, setSelectedPatternName } =
+    props;
+
+  const styles = useStyles2(getStyles);
+  const lang = { grammar: promqlGrammar, name: 'promql' };
+
+  return (
+    <Card className={styles.card}>
+      <Card.Heading>{pattern.name}</Card.Heading>
+      <div className={styles.rawQueryContainer}>
+        <RawQuery
+          aria-label={`${pattern.name} raw query`}
+          query={promQueryModeller.renderQuery({
+            labels: [],
```

Comment:
labels is supposed to be empty?

Reply:
Yep! Ismail suggested we fill them with templates in a comment above, and we can, but currently it looks a little busy. We can add labels here, but they would be just placeholders and not applied to the query.